### PR TITLE
ipvlan: add ipvlan support

### PR DESCRIPTION
**Pull request by ffmancera**
Add ipvlan interface support.

```
- name: ipvlan0
  iface_type: ip_vlan
  state: up
  ip_vlan:
    base_iface: eth1
    mode: l2
    flags: 0
```

Integration test added.

Resolves: https://issues.redhat.com/browse/NMT-1158

### CODE REVIEW

**Comment by ihuguet** [comment]
LGTM

**Comment by ffmancera** [comment]
@cathay4t flags is now an enum, I tested the patch. It should be ready for review. Thank you!

File: src/lib/query/ipvlan.rs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    L2,
+    L3,
+    L3S,
+    Other(u16),
+    Unknown,
+}
+
+impl Default for IpVlanMode {
+    fn default() -> Self {
+        IpVlanMode::Unknown
```
Comment (cathay4t) [inline src/lib/query/ipvlan.rs]: The default should be `IPVLAN_MODE_L3` as kernel code `ipvlan_link_new()` indicate so.

File: src/lib/query/ipvlan.rs [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+}
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone, Copy)]
+#[serde(rename_all = "lowercase")]
+#[non_exhaustive]
+pub enum IpVlanFlag {
+    Private,
+    Vepa,
+    Other(u16),
+    Unknown,
```
Comment (cathay4t) [inline src/lib/query/ipvlan.rs]: I don't think Unknown is required there since we already have Other.

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: src/lib/query/ipvlan.rs (hunk 1 above)
Comment (ffmancera) [reply src/lib/query/ipvlan.rs]: Done, thanks!

**Review by cathay4t — APPROVED** [review]

**Pull request merged**